src/cmd/cli/command/servicemonitor.go

```diff
@@ -2,39 +2,67 @@ package command
 
 import (
 	"context"
+	"errors"
 	"fmt"
 
 	"github.com/DefangLabs/defang/src/pkg/cli"
+	cliClient "github.com/DefangLabs/defang/src/pkg/cli/client"
 	"github.com/DefangLabs/defang/src/pkg/term"
 	defangv1 "github.com/DefangLabs/defang/src/protos/io/defang/v1"
 )
 
+func contextWithServiceStatus(ctx context.Context, targetStatus cli.ServiceStatus, serviceInfos []*defangv1.ServiceInfo) (context.Context, context.CancelFunc) {
+	// New context to make sure tail is only cancelled when message about service status is printed
+	newCtx, cancel := context.WithCancel(context.Background())
+	go func() {
+		err := waitServiceStatus(ctx, targetStatus, serviceInfos)
+		if err == nil {
+			cancel()
+			return
+		}
+
+		if !errors.Is(err, context.Canceled) && !errors.Is(err, cli.ErrDryRun) && !errors.As(err, new(cliClient.ErrNotImplemented)) {
+			term.Info("Service status monitoring failed, we will continue tailing the logs. Press Ctrl+C to detach.")
+		}
+		term.Debugf("failed to wait for service status: %v", err)
```

Comment:
The code would be much simpler if we don't need to log these, but we risk swallow errors for debugging